.github/workflows/maven-build-publish.yml

```diff
@@ -0,0 +1,52 @@
+name: Maven CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  release:
+    types: [published]
+
+jobs:
+  validate:
+    uses: ozone-his/.github/.github/workflows/maven.yml@main
+    with:
+      java-version: 8
+    secrets: inherit
+  
+  spotless:
+    needs: validate
+    uses: ozone-his/.github/.github/workflows/check-spotless.yml@main
+    with:
+      java-version: 8
+
+  # This job publishes snapshots on every push to main
+  pre-release:
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+    needs: validate
+    uses: ozone-his/.github/.github/workflows/publish.yml@main
+    with:
+      java-version: 8
+      maven-args: -DskipTests
+    secrets: inherit
+
+  # This job publishes releases on every tag
+  release:
+    if: ${{ github.event_name == 'release' }}
+    needs: validate
+    uses: ozone-his/.github/.github/workflows/publish.yml@main
+    with:
+      java-version: 8
+      maven-args: -DskipTests
+    secrets: inherit
+
+  # this job notifies OCD3 of a new pre-release(snapshots)
+  notify-ocd3:
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
```

Comment:
Is this guard necessary since it dpends on pre-release? Also, does ocd3 _not_ need to notified of releases?

Reply:
> Is this guard necessary since it depends on pre-release?

No
>  does ocd3 not need to be notified of releases?

Not sure yet?